### ISSUE TITLE
Replica pairs in a volume shouldn't be from the same node

### ISSUE THREAD

**Issue by anushshetty** [heketi/heketi#231]
During volume creation, there is a chance that heketi picks up replica pairs from the same node. This is not a supported RHGS configuration. 

Version-Release number of selected component (if applicable): 1.0.0

How reproducible: Always

Steps to Reproduce:
1. Run heketi server
2. Run heketi client to add nodes and devices
3. Create a volume 

Actual results:

Volume with replica pairs within the same node:

Eg. 
```
Volume Name: vol_f34cf8f3639e6d3a5106d97f8932ec9e
Type: Distributed-Replicate
Volume ID: 9af38984-2b03-4f1e-8691-bf51ad02ac44
Status: Started
Number of Bricks: 4 x 2 = 8
Transport-type: tcp
Bricks:
Brick1: rhshdp05.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_2cd66665852c51bbead1eb76f3b337ad/brick_18dce73c25f8d25fdaca8cff95fb583d/brick
Brick2: rhshdp06.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_546ec7767ff0e27137f56e1c8fb08d61/brick_7fe705e7f45ed1e640b0deb614f59c1f/brick
Brick3: rhshdp06.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_90b022ca94d670b60c16618cb60c73a8/brick_c5e9c2cbcd1579319853360ad9f37951/brick
Brick4: rhshdp05.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_2cd66665852c51bbead1eb76f3b337ad/brick_7bea69193e30120ef2a573535523c0d5/brick
Brick5: rhshdp05.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_2cd66665852c51bbead1eb76f3b337ad/brick_13d3c30116787a45d8a08d5108ef4662/brick
Brick6: rhshdp06.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_546ec7767ff0e27137f56e1c8fb08d61/brick_a8e987488d27d8f498e552f9a80b3f73/brick
Brick7: rhshdp06.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_90b022ca94d670b60c16618cb60c73a8/brick_12691ce59b5a4db67943b064d6b8a311/brick
Brick8: rhshdp06.lab.eng.blr.redhat.com:/var/lib/heketi/mounts/vg_546ec7767ff0e27137f56e1c8fb08d61/brick_4280c648fa95979b3579d30ba8f0b223/brick

```

**Comment by lpabon** [comment]
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1273850

**Comment by lpabon** [comment]
Needs merged for #238
